package-testing/android-relay/app/jacoco.gradle

```diff
@@ -0,0 +1,99 @@
+project.afterEvaluate { project ->
+    setupAndroidReporting()
+}
+
+def setupAndroidReporting() {
+    tasks.withType(Test) {
+        // Whether or not classes without source location should be instrumented
+        jacoco.includeNoLocationClasses = true
+        jacoco.excludes = ['jdk.internal.*']
+    }
+
+    // Grab all build types and product flavors
+    def buildTypes = android.buildTypes.collect { type ->
+        type.name
+    }
+    def productFlavors = android.productFlavors.collect { flavor ->
+        flavor.name
+    }
+    // When no product flavors defined, use empty
+    if (!productFlavors) productFlavors.add('')
```

Comment:
how about enclosing in brackets just for consistency:
```gradle
if (!productFlavors) {
    productFlavors.add('')
}
```

Reply:
dropped jacoco